http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/GKEInterpreter.scala

```diff
@@ -0,0 +1,390 @@
+        else
+          logger.error(s"Create cluster operation has failed for cluster ${dbCluster.id}") >> F.raiseError[Unit](
+            ClusterCreationException(
+              s"Failed to poll cluster creation operation to completion for cluster ${dbCluster.id} and default nodepool ${defaultNodepool.id}"
+            )
+          )
+        getClusterOpt <- gkeService.getCluster(dbCluster.getGkeClusterId)
+        getCluster <- F.fromOption(
+          getClusterOpt,
+          ClusterCreationException(
+            s"Failed to get cluster to retrieve external IP after successful creation polling for ${dbCluster.id}"
+          )
+        )
+        _ <- kubernetesClusterQuery
+          .updateAsyncFields(dbCluster.id,
+                             KubernetesClusterAsyncFields(
+                               IP(getCluster.getEndpoint()),
+                               NetworkFields(
+                                 network,
+                                 subnetwork,
+                                 Config.vpcConfig.subnetworkIpRange
+                               )
+                             ))
+          .transaction
+        _ <- kubernetesClusterQuery.updateStatus(dbCluster.id, KubernetesClusterStatus.Running).transaction
+        _ <- nodepoolQuery.updateStatus(defaultNodepool.id, NodepoolStatus.Running).transaction
+      } yield ()
+
+    createCluster.onError {
+      case e =>
+        for {
+          now <- nowInstant
+          _ <- logger.error(e)(
+            s"Failed to create cluster ${createMessage.clusterId} and nodepool ${createMessage.nodepoolId}"
+          )
+          _ <- kubernetesClusterQuery.updateStatus(createMessage.clusterId, KubernetesClusterStatus.Error).transaction
+          _ <- nodepoolQuery.updateStatus(createMessage.nodepoolId, NodepoolStatus.Error).transaction
+          _ <- appErrorQuery
+            .save(appId, KubernetesError(e.getMessage(), now, ErrorAction.CreateGalaxyApp, ErrorSource.Cluster, None))
+            .transaction
+        } yield ()
+    }
+  }
+
+  def createNodepool(nodepoolId: NodepoolLeoId, appId: AppId)(
+    implicit ev: ApplicativeAsk[F, AppContext]
+  ): F[Unit] = {
+    val createNodepool = for {
+      _ <- logger.info(s"beginning nodepool creation for app ${appId}")
+      dbNodepoolOpt <- nodepoolQuery.getMinimalById(nodepoolId).transaction
+      dbNodepool <- F.fromOption(dbNodepoolOpt, NodepoolNotFoundException(nodepoolId))
+      dbClusterOpt <- kubernetesClusterQuery.getMinimalClusterById(dbNodepool.clusterId).transaction
+      dbCluster <- F.fromOption(
+        dbClusterOpt,
+        KubernetesClusterNotFoundException(s"cluster with id ${dbNodepool.clusterId} not found in database")
+      )
+      req = KubernetesCreateNodepoolRequest(
+        dbCluster.getGkeClusterId,
+        getGoogleNodepool(dbNodepool)
+      )
+      op <- gkeService.createNodepool(req)
```

Comment:
Nodepool probably also needs create-if-not-exists.. any maybe fancier logic depending on how we resolve the concurrency issue.

Reply:
I'd prefer to defer thinking about the new endpoint/discussion in this PR, and merge the base code without concerns for the concurrency issue. 

Having this PR live too long sounds like a bad idea if we want to start parallelizing things like getting the helm client calls in, the IAM calls, claiming nodepools, etc

Reply:
Works for me